distributed/worker.py

```diff
@@ -760,3 +763,41 @@ def apply_function(function, args, kwargs):
     msg['compute_stop'] = end
     msg['thread'] = current_thread().ident
     return msg
+
+
+def convert_args_to_str(args, max_len=1000):
+    """ Convert args to a string, allowing for some arguments to raise
+    exceptions during conversion and ignoring them.
+    """
+    sargs = "("
+    for arg in args:
+        try:
+            sarg = repr(arg)
+        except:
+            sarg = "< could not convert arg to str >"
+        sargs += sarg + ", "
```

Comment:
My consideration was that we want to lazily repr() things, so if we run out of character limit we don't repr redundant things, I guess this can be achieved with length arithmetic as well. I will rewrite.